apps/opencs/view/render/terrainselection.cpp

```diff
@@ -0,0 +1,251 @@
+            int x2 = x * textureSizeToLandSizeModifier + textureSizeToLandSizeModifier + landHeightsNudge;
+            int y1 = y * textureSizeToLandSizeModifier - landHeightsNudge;
+            int y2 = y * textureSizeToLandSizeModifier + textureSizeToLandSizeModifier - landHeightsNudge;
+
+            // Draw straigth edges
+            int landHeightX1Y1 = calculateLandHeight(x1, y1),
+                landHeightX1Y2 = calculateLandHeight(x1, y2),
+                landHeightX2Y1 = calculateLandHeight(x2, y1),
+                landHeightX2Y2 = calculateLandHeight(x2, y2);
+
+            if (north == mSelection.end()) {
+                vertices->push_back(osg::Vec3f(toWorldCoords(x) + nudgeOffset, toWorldCoords(y + 1) - nudgeOffset, landHeightX1Y2+5));
+                vertices->push_back(osg::Vec3f(toWorldCoords(x + 1) + nudgeOffset, toWorldCoords(y + 1) - nudgeOffset, landHeightX2Y2+5));
+            }
+
+            if (south == mSelection.end()) {
+                vertices->push_back(osg::Vec3f(toWorldCoords(x) + nudgeOffset, toWorldCoords(y) - nudgeOffset, landHeightX1Y1+5));
+                vertices->push_back(osg::Vec3f(toWorldCoords(x + 1) + nudgeOffset, toWorldCoords(y) - nudgeOffset, landHeightX2Y1+5));
+            }
+
+            if (east == mSelection.end()) {
+                vertices->push_back(osg::Vec3f(toWorldCoords(x + 1) + nudgeOffset, toWorldCoords(y) - nudgeOffset, landHeightX2Y1+5));
+                vertices->push_back(osg::Vec3f(toWorldCoords(x + 1) + nudgeOffset, toWorldCoords(y + 1) - nudgeOffset, landHeightX2Y2+5));
+            }
+
+            if (west == mSelection.end()) {
+                vertices->push_back(osg::Vec3f(toWorldCoords(x) + nudgeOffset, toWorldCoords(y) - nudgeOffset, landHeightX1Y1+5));
+                vertices->push_back(osg::Vec3f(toWorldCoords(x) + nudgeOffset, toWorldCoords(y + 1) - nudgeOffset, landHeightX1Y2+5));
+            }
+        }
+    }
+
+    newGeometry->setVertexArray(vertices);
+
+    const auto drawArrays = new osg::DrawArrays(osg::PrimitiveSet::LINES);
+
+    drawArrays->setCount(vertices->size());
+
+    newGeometry->addPrimitiveSet(drawArrays);
+
+    mGeode->setDrawable(0, newGeometry);
+}
+
+int CSVRender::TerrainSelection::calculateLandHeight(int x, int y) // global vertex coordinates
+{
+    int cellX(0);
+    int cellY(0);
+    int localX(0);
+    int localY(0);
+
+    if (x >= 0)
+    {
+        cellX = std::floor(x / (landSize - 1));
+        localX = x - cellX * (landSize - 1);
+    }
+    if (y >= 0)
+    {
+        cellY = std::floor(y / (landSize - 1));
+        localY = y - cellY * (landSize - 1);
+    }
+    if (x < 0)
+    {
+        cellX = std::trunc(x / (landSize - 1)) - 1;
+        localX = x - cellX * (landSize - 1);
+    }
+    if (y < 0)
+    {
+        cellY = std::trunc(y / (landSize - 1)) - 1;
+        localY = y - cellY * (landSize - 1);
+    }
+
+    std::string cellId = "#" + std::to_string(cellX) + " " + std::to_string(cellY);
```

Comment:
Please avoid constructing cell IDs manually.

Reply:
There has to be a way to convert global terraintexture-coordiantes x and y to a cell ID. What would the option? Would you prefer all this moved to CellCoordinates class?

Reply:
Yes.